torch/_inductor/runtime/triton_heuristics.py

```diff
@@ -734,6 +734,11 @@ def _precompile_config(self, cfg: Config) -> CompileResult[_KernelType]:
                     ),
                 }
             )
+        if self.device_props.type == "cuda":
+            options.update({
+              "launch_cooperative_grid": compile_meta.get("launch_cooperative_grid", False),
+              "launch_pdl": compile_meta.get("launch_pdl", False),  # True
+            })
```

Comment:
Just for my understanding - programmatic dependent launch allows overlapping of two kernels on the same stream. We are adding the requisite safety guards for the second kernel to be launched with PDL.

Currently we don't have any knowledge of the prior kernel. So we are overlapping preamble of second kernel but necessarily any loads. Similarly we are not overlapping any part of the prior kernel because we do not signal `gdc_launch_dependents`. 

Is this correct ?

Reply:
PDL allows launching the second kernel before the first kernel is fully done, which is when either all blocks of the previous kernel signal the launch or exit. the second kernel has to manually synchronize. it does so here by synchronizing ahead of every load. the follow up will make it be once per kernel.